Release 3.0b1 (#348)
MarketSquare/robotframework-tidy: docs/releasenotes/3.0b1.rst

Example output from the transformers:

::

    *** Keywords ***
    Keyword
        ${var}        Create Resource       ${argument}       value
        Assert        value
        Multi
        ...           line
        ...           args

To read detailed information on each transformer go to the documentation pages:

- `AlignKeywordsSection <https://robotidy.readthedocs.io/en/latest/transformers/AlignKeywordsSection.html>`_
- `AlignTestCasesSection <https://robotidy.readthedocs.io/en/latest/transformers/AlignTestCasesSection.html>`_

Skip formatting
----------------
This feature was added together with align transformers (to help to skip of formatting parts of the code). It is
currently supported by ``AlignKeywordsSection``, ``AlignTestCasesSection`` and ``NormalizeSeparators``
transformers. The support will be extended to other transformers in the upcoming releases.

By using it, it is possible to skip formatting of the code with given type or values. It is also possible
by defining ``# robotidy: off`` disablers in the code, but with skip formatting you can disable formatting on all
instances of code that match your conditions.

> For example, you can have the keyword that you want to to align automatically (using ``AlignKeywordsSection``) and

Reviewer: ```suggestion
For example, you can have the keyword that you want to align automatically (using ``AlignKeywordsSection``) and
```

IndentNestedKeywords - AND handling
------------------------------------
``indent_and`` parameter in ``IndentNestedKeywords`` now allows you to configure one of three ways of handling the
``AND`` argument used in ``Run Keywords`` (`#328 <https://github.com/MarketSquare/robotframework-tidy/issues/328>`_).

``indent_and=split`` (default, previously False)

::

    *** Keywords ***
        Keyword
            Run Keywords
            ...    No Operation  arg
            ...    AND
            ...    No Operation  arg1  arg2

``indent_and=split_and_indent`` (previously True)

::

    *** Keywords ***
        Keyword
            Run Keywords
            ...        No Operation  arg
            ...    AND
            ...        No Operation  arg1  arg2

``indent_and=keep_in_line`` (new mode)

::

*** Keywords ***
    Keyword
        Run Keywords
        ...    No Operation  arg    AND
        ...    No Operation  arg1  arg2

Continuation indent
--------------------
It's extension of ``--indent`` option added in ``2.4.0`` release. ``--indent`` allows you to configure indentation

> separetely from separator (``--spacecount``). Continuation indent ``--continuation-indent`` allow you define indent

Reviewer: ```suggestion
separately from separator (``--spacecount``). Continuation indent ``--continuation-indent`` allows you define indent
```

after `...` continuation marks (`#331 <https://github.com/MarketSquare/robotframework-tidy/issues/331>`_).

Following configuration::

    robotidy --spacecount 2 --indent 4

leads to this formatting:

::

*** Test Cases ***
Test case
    Multiline  ${argument}
    ...  value
    ...  value2  value3  value4

By using ``--continuation-indent``::

    robotidy --spacecount 2 --indent 4 --continuation-indent 4

we can achieve this:

::

*** Test Cases ***
Test case
    Multiline  ${argument}
    ...    value
    ...    value2  value3  value4

Fixes
-----

> - documentation now properly states that ``NormalizeTags`` is default transformer (`#335 <https://github.com/MarketSquare/robotframework-tidy/issues/335>`_)

Reviewer: ```suggestion
- documentation now properly states that ``NormalizeTags`` is the default transformer (`#335 <https://github.com/MarketSquare/robotframework-tidy/issues/335>`_)
```

- default values for some options in help message are now formatted correctly (`#337 <https://github.com/MarketSquare/robotframework-tidy/issues/337>`_)

Other
-----
There are major changes to core features of Robotidy that doesn't affect the users but make it easier to develop and
maintain the tool:

- Configuration is now stored in separate class (`#329 <https://github.com/MarketSquare/robotframework-tidy/issues/329>`_)

> - Transformers inherits now from Robotidy internal ``Transformer`` class rather than ``ModelTransformer``

Reviewer: ```suggestion
- Transformers inherit now from Robotidy internal ``Transformer`` class rather than ``ModelTransformer``
```
